### PR TITLE
Balancer not working

### PR DESCRIPTION
**Pull request by rafaeljesus**
Fix load balancer not working

### CODE REVIEW

File: pkg/proxy/register.go [hunk 1]
```diff
@@ -42,10 +42,22 @@ func (p *Register) AddMany(routes []*Route) error {
...
 
+	var balancer Balancer
+	if definition.IsBalancerDefined() {
+		log.WithField("balancing_alg", definition.Upstreams.Balancing).Debug("Using a load balancing algorithm")
+
+		var err error
+		balancer, err = NewBalancer(definition.Upstreams.Balancing)
+		if err != nil {
+			log.WithError(err).Error("Could not create a balancer")
+			return err
```
Comment (italolelis) [inline pkg/proxy/register.go]: I would do `errors.Wrap(err, "Could not create a balancer")`

File: pkg/proxy/register.go [hunk 2]
```diff
@@ -58,45 +70,33 @@ func (p *Register) Add(route *Route) error {
...
 		// but currently we have several points of definition creation
-		if proxyDefinition.Upstreams != nil && proxyDefinition.Upstreams.Targets != nil && len(proxyDefinition.Upstreams.Targets) > 0 {
-			log.WithField("balancing_alg", proxyDefinition.Upstreams.Balancing).Debug("Using a load balancing algorithm")
-			balancer, err := NewBalancer(proxyDefinition.Upstreams.Balancing)
-			if err != nil {
-				log.WithError(err).Error("Could not create a balancer")
-				return
-			}
-
+		if proxyDefinition.IsBalancerDefined() {
```
Comment (italolelis) [inline pkg/proxy/register.go]: I would just check for `if balancer != nil {` since you set the balancer before

File: pkg/proxy/register.go [hunk 3]
```diff
@@ -58,45 +70,33 @@ func (p *Register) Add(route *Route) error {
...
 }
 
-func (p *Register) createDirector(proxyDefinition *Definition) func(req *http.Request) {
+func (p *Register) createDirector(proxyDefinition *Definition, balancer Balancer) func(req *http.Request) {
 	return func(req *http.Request) {
-		var target *url.URL
-		var err error
+		var upstreamURL string
 
 		// TODO: find better solution
```
Comment (italolelis) [inline pkg/proxy/register.go]: Is this comment still necessary?

**Review by italolelis — CHANGES_REQUESTED** [review]

File: pkg/proxy/register.go (hunk 1 above)
Comment (italolelis) [reply pkg/proxy/register.go]: And from `"github.com/pkg/errors"`

**Comment by vgarvardt** [comment]
👍

**Review by italolelis — APPROVED** [review]

**Pull request merged**